libs/libcommon/src/libcommon/queue.py

```diff
@@ -274,6 +277,11 @@ def __exit__(
         self.release()
         return False
 
+    @classmethod
+    def git_rev(cls, dataset: str, revision: str, job_id: str, sleeps: Sequence[float] = _default_sleeps) -> "lock":
```

Comment:
much better.

Two requests:
1. add docstring (there are many arguments, and revision is not clear: main ? the commit ? the ref (refs/convert/parquet) <- I know that it's this one, but it's not clear) maybe "branch" or "ref", because it's where we will push.
2. I think the name could be clearer (and btw: I think we should avoid abbreviations like rev, I agree with https://github.com/huggingface/moon-landing/pull/5339/files <- every variable/function should ideally be made of valid words)